### PR TITLE
Make network plugin a variable to enable BYOCNI installs

### PR DESCRIPTION
**Pull request by hhoover**
this PR turns the network plugin into a variable so you can specify "none" for BYOCNI installs.

Also adds an example for how to use this module, in a separate commit if it's not needed.

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -89,3 +89,9 @@ variable "subnet_id" {
   description = "The ID of the subnet where to place the node pool."
   type        = string
 }
+
+variable "network_plugin" {
+  description = "The Network plugin used for Azure"
```
Comment (bmcustodio) [inline variables.tf:94]:
```suggestion
  description = "The network plugin to use (one of 'azure' or 'none')."
```

File: variables.tf [hunk 2]
```diff
@@ -89,3 +89,9 @@ variable "subnet_id" {
...
+  default = "azure"
```
Comment (bmcustodio) [inline variables.tf:95]: Nit: move up one line for alphabetical ordering.

File: README.md [hunk 3]
```diff
@@ -45,6 +45,7 @@ An opinionated Terraform module that can be used to create and manage an AKS clu
 | <a name="input_max_nodes"></a> [max\_nodes](#input\_max\_nodes) | The maximum number of nodes in the AKS cluster. | `number` | `4` | no |
 | <a name="input_min_nodes"></a> [min\_nodes](#input\_min\_nodes) | The minimum number of nodes in the AKS cluster. | `number` | `3` | no |
 | <a name="input_name"></a> [name](#input\_name) | The name of the AKS cluster. | `string` | n/a | yes |
+| <a name="input_network_plugin"></a> [network\_plugin](#input\_network\_plugin) | The Network plugin used for Azure | `string` | `"azure"` | no |
```
Comment (bmcustodio) [inline README.md:48]: Can you please re-run `make docs` after making the suggested change?

File: example/aks.tf [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  region              = var.region
+  resource_group_name = basename(azurerm_resource_group.arg.id)
+  service_cidr        = var.service_cidr
+  subnet_id           = module.network.vnet_subnets[0]
+  network_plugin      = "none"
+}
+
+// Provision the AKS cluster.
+module "provisioner" {
+  source = "git@github.com:isovalent/terraform-k8s-cilium.git?ref=v1.0"
```
Comment (bmcustodio) [inline example/aks.tf:34]:
```suggestion
  source = "git::https://github.com/isovalent/terraform-k8s-cilium.git?ref=v1.0"
```

File: example/aks.tf [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+  cilium_helm_chart            = "cilium/cilium"
+  cilium_helm_version          = "1.12.2"
+  cilium_helm_values_file_path = "${abspath(path.module)}/cilium-helm-values.yaml"
+  cilium_namespace             = "cilium"
+  ipsec_key               = var.ipsec_key
```
Comment (bmcustodio) [inline example/aks.tf:40]: Missing some `terraform fmt` 😉

File: example/variables.tf [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+  type        = string
+}
+
+variable "service_cidr" {
+  default     = "10.0.0.0/16"
+  description = "The CIDR to use for 'ClusterIP' services."
+  type        = string
+}
+
+variable "subscription_id" {
```
Comment (bmcustodio) [inline example/variables.tf:57]: Please remove this variable and the next one 🤐

**Review by bmcustodio — CHANGES_REQUESTED** [review]
Looking great but I have some requests 🙏🏼

File: variables.tf (hunk 1 above)
Comment (hhoover) [reply variables.tf]: fixed

File: example/variables.tf (hunk 6 above)
Comment (hhoover) [reply example/variables.tf:57]: this is an example, made to "just work".

**Comment by darox** [comment]
@hhoover @bmcustodio shall I merge it and cut a new release?

**Pull request merged**